base_tier_validation_server_action/models/tier_review.py

```diff
@@ -7,24 +7,33 @@
 class TierReview(models.Model):
     _inherit = "tier.review"
 
+    def _run_server_action(self):
+        for rec in self.filtered(
+            lambda review: not review.record_id.tier_validation_before_write
```

Comment:
this will loop twice, can we skip records in the loop itself?

Reply:
@renda-dev Do you have any thoughts on this?

Reply:
I think that is not a really big deal, as long as it's not recursive and there are not too many operations inside the loop